From 03e976000158500ac918e2023b057b49e4753779 Mon Sep 17 00:00:00 2001
From: Yufen Yu <yuyufen@huawei.com>
Date: Fri, 15 Mar 2019 10:49:47 +0200
Subject: [PATCH 2/2] block: Update[v3] (Add a new flag BLK_MQ_POLL_CLASSIC for
 hybrid poll)

For q->poll_nsec == -1, means doing classic poll, not hybrid poll.
We introduce a new flag BLK_MQ_POLL_CLASSIC to replace -1, which
may make code much easier to read.

Signed-off-by: Yufen Yu <yuyufen@huawei.com>
---
 block/blk-mq.c    | 4 +---
 block/blk-sysfs.c | 4 ++--
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 699ff704ca6a..76a3f78c566a 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -3349,8 +3349,6 @@ static bool blk_mq_poll_hybrid_sleep(struct request_queue *q,
 	 *  0:	use half of prev avg
 	 * >0:	use this specific value
 	 */
-	if (q->poll_nsec == BLK_MQ_POLL_CLASSIC)
-		return false;
 	if (q->poll_nsec > 0)
 		nsecs = q->poll_nsec;
 	else
@@ -3393,7 +3391,7 @@ static bool blk_mq_poll_hybrid(struct request_queue *q,
 {
 	struct request *rq;
 
-	if (q->poll_nsec == -1)
+	if (q->poll_nsec == BLK_MQ_POLL_CLASSIC)
 		return false;
 
 	if (!blk_qc_t_is_internal(cookie))
diff --git a/block/blk-sysfs.c b/block/blk-sysfs.c
index 615f5cc1ee45..cd127067998a 100644
--- a/block/blk-sysfs.c
+++ b/block/blk-sysfs.c
@@ -361,7 +361,7 @@ static ssize_t queue_poll_delay_show(struct request_queue *q, char *page)
 	int val;
 
 	if (q->poll_nsec == BLK_MQ_POLL_CLASSIC)
-		val = -1;
+		val = BLK_MQ_POLL_CLASSIC;
 	else
 		val = q->poll_nsec / 1000;
 
@@ -380,7 +380,7 @@ static ssize_t queue_poll_delay_store(struct request_queue *q, const char *page,
 	if (err < 0)
 		return err;
 
-	if (val == -1)
+	if (val == BLK_MQ_POLL_CLASSIC)
 		q->poll_nsec = BLK_MQ_POLL_CLASSIC;
 	else
 		q->poll_nsec = val * 1000;
-- 
2.21.0.155.ge902e9bcae

